engine/table/src/main/java/io/deephaven/engine/table/impl/by/ChunkedOperatorAggregationHelper.java

```diff
@@ -242,7 +242,11 @@ private static QueryTable aggregation(
         final QueryTable result = new QueryTable(resultRowSet, resultColumnSourceMap);
         ac.propagateInitialStateToOperators(result, outputPosition.intValue());
 
-        if (input.isRefreshing()) {
+        if (!input.isRefreshing()) {
+            if (resultRowSet.isFlat()) {
+                result.setFlat();
```

Comment:
We can probably always set the result of no-key agg to flat. And the result of ticking agg, if input is add-only, append-only, blink, or if preserve-empty is true.